concent_api/core/models.py

```diff
@@ -494,6 +494,70 @@ def clean(self) -> None:
             raise ValidationError({
                 'computation_deadline': "TaskToCompute deadline mismatch"
             })
+        # Validation for every nested message which is stored in Control database
+        # Every nested message must be the same as message stored separately.
+        if deserialized_report_computed_task.task_to_compute != deserialized_task_to_compute:
```

Comment:
Generally looks ok but I think maybe this can be replaces with some generic function with parameters like `message_1=None, message_2=None, message_1_to_deserialize=None, message_2_to_deserialize=None` (either `message_1` or `message_1_to_deserialize` will be passed and then function will now if it should extra deserialize it) Generating error message should be possible to do generic way. And then prepare some data structure (dictionary could do) of which messages should be equal to each and go through it.

Reply:
Change to new solution.